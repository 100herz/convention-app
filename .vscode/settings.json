{
  "editor.codeActionsOnSave": {
    "source.fixAll": true
  },
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnPaste": true,
  "editor.formatOnSave": true,

  "files.exclude": {
    "**/.git": true,
    "**/.DS_Store": true
  },
  "files.watcherExclude": {
    "**/node_modules/**": true
  },

  // Extensions
  // ----------

  // Jest
  "jest.rootPath": "src/",
  "jest.showCoverageOnLoad": true,
  "jest.enableSnapshotUpdateMessages": false
}
