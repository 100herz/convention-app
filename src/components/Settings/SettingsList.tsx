import React, { useState } from 'react'
import { StyleSheet, View } from 'react-native'

import Text from '@components/UI/Text'
import Switch from '@components/UI/Switch'
import { Setting } from '@models/setting'
import { DefaultStyles, defaultStyles } from '@styles/theme'

interface Props {
  data: Setting[]
}

const LinkList: React.FC<Props> = ({ data }) => {
  const [isEnabled, setIsEnabled] = useState(false)

  const toggleSwitch = () => setIsEnabled(previousState => !previousState)

  return (
    <View style={styles.listContainer}>
      {data.map(item => (
        <View style={styles.listItemContainer} key={item.id}>
          <Text>{item.name}</Text>
          <Switch style={styles.switch} onValueChange={toggleSwitch} value={isEnabled} testID={`switch-${item.id}`} />
        </View>
      ))}
    </View>
  )
}

type Styles = DefaultStyles

const styles = StyleSheet.create<Styles>({
  ...defaultStyles,
  listItemContainer: {
    ...defaultStyles.listItemContainer,
    marginRight: -8,
    paddingVertical: 5,
  },
})

export default LinkList
