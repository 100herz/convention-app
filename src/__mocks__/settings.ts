import { Setting } from '@models/settings'

export const settingItem: Setting = { id: 1, name: 'Push Mitteilungen' }

export const settingArray: Setting[] = [{ id: 1, name: 'Push Mitteilungen' }]
