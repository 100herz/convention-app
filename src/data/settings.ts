import { Legal, LegalAddition, Setting } from '@models/settings'

export const legalScreens: Legal[] = [
  {
    id: 79,
    title: { rendered: 'Impressum' },
    content: { rendered: '' },
  },
  {
    id: 1105,
    title: { rendered: 'Datenschutz' },
    content: { rendered: '' },
  },
]

export const legalAddition: LegalAddition = {
  id: 1,
  name: 'Anfrage stellen',
  href: 'mailto:anfrage@convention-net.de',
}

export const settingArray: Setting[] = [
  {
    id: 1,
    name: 'Push Mitteilungen',
  },
]
