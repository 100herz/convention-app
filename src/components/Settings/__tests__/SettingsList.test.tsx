import React from 'react'
import { act, fireEvent, render } from '@testing-library/react-native'

import SettingsList from '../SettingsList'
import { settingArray } from '@__mocks__/settings'

describe('<SettingsList />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<SettingsList data={settingArray} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('sets the value correctly', () => {
    const { getByTestId } = render(<SettingsList data={settingArray} />)

    act(() => {
      fireEvent(getByTestId('switch-1'), 'valueChange')
    })

    // TODO: Add the expect
    // expect(toggleSwitch).toHaveBeenCalledTimes(1)
  })
})
