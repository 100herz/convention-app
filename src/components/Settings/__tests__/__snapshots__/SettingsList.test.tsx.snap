// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsList /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginVertical": 10,
      "paddingHorizontal": 25,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#dfe3e8",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginRight": -8,
        "paddingHorizontal": 15,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "fontFamily": "sans",
          "fontSize": 15,
          "lineHeight": 22.5,
        }
      }
    >
      Push Mitteilungen
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="switch-1"
    >
      <Text>
        false
      </Text>
    </View>
  </View>
</View>
`;
